src/Microsoft.SqlTools.ServiceLayer/Scripting/ScriptingService.cs

```diff
@@ -133,14 +187,31 @@ public void InitializeService(ServiceHost serviceHost)
         {
             try
             {
-                ScriptingScriptOperation operation = new ScriptingScriptOperation(parameters);
-                operation.PlanNotification += (sender, e) => this.SendEvent(requestContext, ScriptingPlanNotificationEvent.Type, e);
-                operation.ProgressNotification += (sender, e) => this.SendEvent(requestContext, ScriptingProgressNotificationEvent.Type, e);
-                operation.CompleteNotification += (sender, e) => this.SendEvent(requestContext, ScriptingCompleteEvent.Type, e);
+                // convert owner uri received from parameters to lookup for its
+                // associated connection and build a connection string out of it
+                ConnectionInfo connInfo;
+                ScriptingService.ConnectionServiceInstance.TryFindConnection(
+                    parameters.OwnerUri,
+                    out connInfo);
+                if (connInfo != null)
+                {
+                    parameters.ConnectionString = ConnectionService.BuildConnectionString(connInfo.ConnectionDetails);
+                }
 
-                RunTask(requestContext, operation);
+                // if the scripting operation is for select
```

Comment:
>/ if the scripting operation is for select [](start = 17, length = 42)

Please add a comment how "ScriptSelect" although async, wait for the script result to return, where as other scripting operations spawn a task for the scripting operation, but return immediately.

Reply:
Or make it spawn a task to be consistent.

---
In reply to: [141987833](https://github.com/Microsoft/sqltoolsservice/pull/446#discussion_r141987833) [](ancestors = 141987833)